src/screens/shared/auth/SignUp.js

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+
+import { Button } from '@material-ui/core';
+
+import Dropdown from '@components/Dropdown';
+import FieldInput from '@components/FieldInput';
+import '@styles/SignUp.css';
+import { validateField, updateUserFields } from '@lib/utils';
+
+class SignUp extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+      userTypes: '',
+      groupGapContact: false,
+      errors: {}
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.createAccount = this.createAccount.bind(this);
+  }
+
+  handleChange(name) {
+    return event => {
+      this.setState({
+        [name]: event.target.value
+      });
+    };
+  }
+
+  handleDropdownChange(options) {
+    return event => {
+      this.setState({
+        userTypesIndex: event.target.value,
+        userTypes: options[event.target.value]
+      });
+    };
+  }
+
+  async createAccount() {
+    // Keep track of whether we've found any errors
+    let foundErrors = false;
+
+    // For each field in this onboarding step, validate, and add to errors object
+    const fieldsToValidate = [
+      'firstName',
+      'lastName',
+      'email',
+      'userTypes',
+      'password'
+    ];
+
+    const { state } = this;
+    const allErrorMessages = await Promise.all(
+      fieldsToValidate.map(f => validateField(f, state[f]))
+    ).catch(e => {
+      console.error(e);
+    });
+
+    const newErrors = {};
+    fieldsToValidate.forEach((field, i) => {
+      const errorMessage = allErrorMessages[i];
+      if (errorMessage !== '') {
+        newErrors[field] = errorMessage;
```

Comment:
![image](https://user-images.githubusercontent.com/35586281/114113249-0a93b980-9893-11eb-83b8-2676cf36bd14.png)

getting this error when i submit